go.mod

```diff
@@ -1,140 +1,158 @@
 
-go 1.21
+go 1.22.4
+
+toolchain go1.22.5
 
 require (
-	cloud.google.com/go/bigquery v1.59.1
-	cloud.google.com/go/firestore v1.14.0
-	cloud.google.com/go/pubsub v1.36.1
-	cloud.google.com/go/storage v1.37.0
+	cloud.google.com/go/bigquery v1.61.0
+	cloud.google.com/go/firestore v1.15.0
+	cloud.google.com/go/pubsub v1.38.0
+	cloud.google.com/go/storage v1.42.0
 	firebase.google.com/go v3.13.0+incompatible
 	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.11.1
 	github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.6.0
 	github.com/Azure/azure-sdk-for-go/sdk/storage/azblob v1.0.0
 	github.com/alpacahq/alpaca-trade-api-go/v2 v2.8.0
 	github.com/apache/arrow/go/v17 v17.0.0-20240521152544-34f042762061
-	github.com/aws/aws-sdk-go v1.44.192
-	github.com/aws/aws-sdk-go-v2 v1.18.1
-	github.com/aws/aws-sdk-go-v2/config v1.18.19
-	github.com/aws/aws-sdk-go-v2/credentials v1.13.18
+	github.com/aws/aws-sdk-go v1.54.2
+	github.com/aws/aws-sdk-go-v2 v1.20.3
+	github.com/aws/aws-sdk-go-v2/config v1.18.21
+	github.com/aws/aws-sdk-go-v2/credentials v1.13.20
 	github.com/aws/aws-sdk-go-v2/feature/dynamodb/attributevalue v1.10.29
 	github.com/aws/aws-sdk-go-v2/feature/s3/manager v1.11.59
-	github.com/aws/aws-sdk-go-v2/service/dynamodb v1.19.11
+	github.com/aws/aws-sdk-go-v2/service/dynamodb v1.21.4
 	github.com/aws/aws-sdk-go-v2/service/dynamodbstreams v1.14.14
-	github.com/aws/aws-sdk-go-v2/service/s3 v1.31.0
-	github.com/aws/smithy-go v1.13.5
+	github.com/aws/aws-sdk-go-v2/service/s3 v1.32.0
+	github.com/aws/smithy-go v1.20.2
 	github.com/bradleyjkemp/cupaloy v2.3.0+incompatible
-	github.com/cespare/xxhash/v2 v2.2.0
+	github.com/cespare/xxhash/v2 v2.3.0
 	github.com/databricks/databricks-sdk-go v0.41.0
 	github.com/databricks/databricks-sql-go v1.5.5
 	github.com/dropbox/dropbox-sdk-go-unofficial/v6 v6.0.5
 	github.com/elastic/go-elasticsearch/v8 v8.9.0
-	github.com/estuary/flow v0.3.12-0.20240118233153-7681c66d4e1f
-	github.com/evanphx/json-patch/v5 v5.6.0
-	github.com/go-mysql-org/go-mysql v1.5.0
-	github.com/go-sql-driver/mysql v1.6.0
+	github.com/estuary/flow v0.4.0
+	github.com/evanphx/json-patch/v5 v5.9.0
+	github.com/go-mysql-org/go-mysql v1.8.0
+	github.com/go-sql-driver/mysql v1.8.1
 	github.com/gogo/protobuf v1.3.2
 	github.com/golang-jwt/jwt/v5 v5.2.1
 	github.com/google/uuid v1.6.0
-	github.com/iancoleman/orderedmap v0.2.0
+	github.com/iancoleman/orderedmap v0.3.0
 	github.com/invopop/jsonschema v0.5.0
+	github.com/jackc/pgconn v1.14.3
 	github.com/jackc/pglogrepl v0.0.0-20240307033717-828fbfe908e9
-	github.com/jackc/pgx/v4 v4.14.0
+	github.com/jackc/pgx/v4 v4.18.3
 	github.com/jackc/pgx/v5 v5.6.0
-	github.com/jmoiron/sqlx v1.3.3
-	github.com/klauspost/compress v1.17.8
-	github.com/marcboeker/go-duckdb v1.7.0
+	github.com/jmoiron/sqlx v1.4.0
+	github.com/klauspost/compress v1.17.9
+	github.com/marcboeker/go-duckdb v1.5.6
 	github.com/mattn/go-sqlite3 v2.0.3+incompatible
-	github.com/microsoft/go-mssqldb v0.19.0
+	github.com/mdibaiee/vitess v0.15.7
```

Comment:
Vitess is pretty large, although the sqlparser part may be smaller. It may be possible to extract the sqlparser part only. What we could do is fork vitess into an estuary repository, and strip it down to the sqlparser only.

Note: you will never lose access to my fork even if I make it private, see: https://trufflesecurity.com/blog/anyone-can-access-deleted-and-private-repo-data-github

Reply:
I tried stripping the repo down to sqlparser, but sqlparser depends on so many packages in the repository, and they subsequently depend on others, so I left the repo as-is but forked it by estuary: github.com/estuary/vitess